backup/queries_relations.go

```diff
@@ -115,15 +127,27 @@ func getUserTableRelationsWithIncludeFiltering(connectionPool *dbconn.DBConn, in
 	includeOids := getOidsFromRelationList(connectionPool, includedRelationsQuoted)
 	oidStr := strings.Join(includeOids, ", ")
 	query := fmt.Sprintf(`
-	SELECT n.oid AS schemaoid,
-		c.oid AS oid,
-		quote_ident(n.nspname) AS schema,
-		quote_ident(c.relname) AS name
-	FROM pg_class c
+	SELECT schemaoid, oid, schema, name FROM (
+		SELECT n.oid AS schemaoid,
+			c.oid AS oid,
+			quote_ident(n.nspname) AS schema,
+			quote_ident(c.relname) AS name,
+			coalesce(prt.pages, c.relpages) AS pages
+		FROM pg_class c
 		JOIN pg_namespace n ON c.relnamespace = n.oid
-	WHERE c.oid IN (%s)
+		LEFT JOIN (
+			SELECT
+				p.parrelid,
+				sum(pc.relpages) AS pages
+			FROM pg_partition_rule AS pr
+			JOIN pg_partition AS p ON pr.paroid = p.oid
+			JOIN pg_class AS pc ON pr.parchildrelid = pc.oid
+			GROUP BY p.parrelid
```

Comment:
It can be a little bit faster to execute this, and possibly another, query, if we can duplicate `WHERE c.oid IN (%s)` for partitioned tables subquery. Current plan, which shows no filtering optimization for `prt` subquery, looks like this:
```
                                                     QUERY PLAN                                                     
--------------------------------------------------------------------------------------------------------------------
 Sort  (cost=34.60..34.61 rows=4 width=148)
   Sort Key: (COALESCE(prt.pages, (c.relpages)::bigint)), c.oid
   ->  Hash Left Join  (cost=18.33..34.57 rows=4 width=148)
         Hash Cond: (c.oid = prt.parrelid)
         ->  Hash Join  (cost=1.16..17.30 rows=4 width=140)
               Hash Cond: (c.relnamespace = n.oid)
               ->  Seq Scan on pg_class c  (cost=0.00..16.11 rows=2 width=76)
                     Filter: ((relkind = 'r'::"char") AND (oid = ANY ('{1,2,3}'::oid[])))
               ->  Hash  (cost=1.07..1.07 rows=7 width=68)
                     ->  Seq Scan on pg_namespace n  (cost=0.00..1.07 rows=7 width=68)
         ->  Hash  (cost=17.16..17.16 rows=1 width=12)
               ->  Subquery Scan on prt  (cost=17.14..17.16 rows=1 width=12)
                     ->  HashAggregate  (cost=17.14..17.15 rows=1 width=12)
                           Group Key: p.parrelid
                           ->  Hash Join  (cost=2.03..17.10 rows=9 width=8)
                                 Hash Cond: (pr.paroid = p.oid)
                                 ->  Hash Join  (cost=1.01..15.96 rows=9 width=8)
                                       Hash Cond: (pc.oid = pr.parchildrelid)
                                       ->  Seq Scan on pg_class pc  (cost=0.00..12.99 rows=499 width=8)
                                       ->  Hash  (cost=1.00..1.00 rows=1 width=8)
                                             ->  Seq Scan on pg_partition_rule pr  (cost=0.00..1.00 rows=1 width=8)
                                 ->  Hash  (cost=1.00..1.00 rows=1 width=8)
                                       ->  Seq Scan on pg_partition p  (cost=0.00..1.00 rows=1 width=8)
 Optimizer: Postgres query optimizer
(24 rows)

```
But such optimization possibly makes little sense here, so I approve.

Reply:
So, you mean the following, don't you?
```SELECT schemaoid, oid, schema, name FROM (
		SELECT n.oid AS schemaoid,
			c.oid AS oid,
			quote_ident(n.nspname) AS schema,
			quote_ident(c.relname) AS name,
			coalesce(prt.pages, c.relpages) AS pages
		FROM pg_class c
		JOIN pg_namespace n ON c.relnamespace = n.oid
		LEFT JOIN (
			SELECT
				p.parrelid,
				sum(pc.relpages) AS pages
			FROM pg_partition_rule AS pr
			JOIN pg_partition AS p ON pr.paroid = p.oid
			JOIN pg_class AS pc ON pr.parchildrelid = pc.oid
			--- changes start here
                         WHERE p.parrelid IN (%s)
                        --- changes end here
			GROUP BY p.parrelid
		) AS prt ON prt.parrelid = c.oid
		WHERE c.oid IN (%s)
		AND (relkind = 'r')
	) res
	ORDER BY pages DESC, oid

Reply:
Right